pmd-core/src/main/java/net/sourceforge/pmd/AbstractPropertyDescriptorFactory.java

```diff
@@ -28,30 +30,42 @@
  */
 public abstract class AbstractPropertyDescriptorFactory<T> implements PropertyDescriptorFactory<T> {
 
-    protected static final Map<PropertyDescriptorField, Boolean> CORE_FIELD_TYPES_BY_KEY
-        = CollectionUtil.mapFrom(new PropertyDescriptorField[] {NAME, DESCRIPTION, DEFAULT_VALUE, DELIMITER},
-                                 new Boolean[] {true, true, true, false});
+
+
+
+    static {
+        System.err.println(CORE_FIELD_TYPES_BY_KEY);
```

Comment:
I have a tool written, needs some polish yet.  But here's what it told me when I said find a path from `IntegerProperty` to `PropertyDescriptorUtil`, obviously for the version of the code with the problem:
```
INITIAL_CLASS_LOAD: net/sourceforge/pmd/lang/rule/properties/IntegerProperty.<clinit>()V
SUPERCLASS: net/sourceforge/pmd/lang/rule/properties/AbstractNumericProperty.<clinit>()V
SUPERCLASS: net/sourceforge/pmd/lang/rule/properties/AbstractSingleValueProperty.<clinit>()V
SUPERCLASS: net/sourceforge/pmd/lang/rule/properties/AbstractProperty.<clinit>()V
INTERFACE: net/sourceforge/pmd/PropertyDescriptor.<clinit>()V
INVOKE_METHOD: net/sourceforge/pmd/lang/rule/properties/PropertyDescriptorUtil$ExpectedFieldsBuilder.get()Ljava/util/Map;
```

This explains why pushing `ExpectedFieldsBuilder` out of `PropertyDescriptorUtil` can break the cycle.